src/order/boolean_algebra.lean

```diff
@@ -159,9 +159,17 @@ by rw [sup_comm, sup_sdiff_self_right, sup_comm]
 lemma sup_sdiff_symm : x ⊔ (y \ x) = y ⊔ (x \ y) :=
 by rw [sup_sdiff_self_right, sup_sdiff_self_right, sup_comm]
 
-lemma sup_sdiff_of_le (h : x ≤ y) : x ⊔ (y \ x) = y :=
+lemma sup_sdiff_cancel_right (h : x ≤ y) : x ⊔ (y \ x) = y :=
 by conv_rhs { rw [←sup_inf_sdiff y x, inf_eq_right.2 h] }
 
+lemma sdiff_sup_cancel (h : y ≤ x) : x \ y ⊔ y = x := by rw [sup_comm, sup_sdiff_cancel_right h]
```

Comment:
These two lemmas have the same LHS as `sup_sdiff_self_right` and `sup_sdiff_self_left`; so I think the second one should have `left` in the name too.

Reply:
```suggestion
lemma sup_sdiff_cancel_right (h : x ≤ y) : x ⊔ (y \ x) = y :=
by conv_rhs { rw [←sup_inf_sdiff y x, inf_eq_right.2 h] }

lemma sdiff_sup_cancel_left (h : x ≤ y) : (y \ x) ⊔ x = y := by rw [sup_comm, sup_sdiff_cancel_right h]
```

Reply:
I got the names from the add-sub versions. I think the boolean algebra API has broken lemma names (I personally can never guess the correct lemma) and I'm planning to rename most of them.

Reply:
Let's fix those names in another PR